### PR TITLE
Fixed `apiType` and broken error handling

### PR DESCRIPTION
**Pull request by kirrg001**
**PR contains a couple of clean commits.** Please look at each commit individually 👍 

A couple of regression tests were broken, which made me curious.

1. API v2 pipeline changes the error handler, which broke the error handling for the site, because it expects an error and not an object with attrs. If you add a catch handler, you expect to get an error and not an object. 

2. `apiType` was always undefined for functional calls e.g. `api.v2.settings.edit({..} {context: {internal:true}})` returned true for`isContentAPI()`. The `apiType` was only handled on http layer, not on functional layer.

**Please use "rebase & commit"!**

- [ ] review & test @allouis 
- [x] review & test @gargol 
- [x] @gargol Kevin pushed some mobiledoc/html changes (https://github.com/TryGhost/Ghost/commit/4f9e687f62e0f1023c1d10aa6b06d82c523fc2a6), which breaks some regression importer & posts tests. Can you please ask him if they are okay to ignore for now?

### CODE REVIEW

File: core/server/api/shared/pipeline.js [hunk 1]
```diff
@@ -156,9 +159,6 @@ const pipeline = (apiController, apiUtils) => {
                 })
                 .then(() => {
                     return frame.response;
-                })
-                .catch((err) => {
-                    throw {err, docName, method};
```
Comment (kirrg001) [inline core/server/api/shared/pipeline.js]: @gargol Which consequences has this change on the error message **right now**? e.g. the site uses browse & read only, which most common error is a 404.

File: core/server/api/v2/index.js [hunk 2]
```diff
@@ -103,10 +103,6 @@ module.exports = {
         return shared.pipeline(require('./slack'), localUtils);
```
Comment (kirrg001) [inline core/server/api/v2/index.js]: Every content API ctrl uses the pattern `{resource}-public`, except of this ctrl. Consistency change.

File: core/server/api/shared/frame.js [hunk 3]
```diff
@@ -10,7 +10,7 @@ class Frame {
         this.user = {};
```
Comment (kirrg001) [inline core/server/api/shared/frame.js]: @allouis Please review & test as good as possible 👍

File: core/server/api/v2/index.js [hunk 4]
```diff
@@ -119,15 +103,39 @@ module.exports = {
...
+     */
+    get pagesPublic() {
+        return shared.pipeline(require('./pages-public'), localUtils, 'content');
+    },
+
+    get tagsPublic() {
+        return shared.pipeline(require('./tags-public'), localUtils, 'content');
+    },
+
+    get publicSettings() {
```
Comment (kirrg001) [inline core/server/api/v2/index.js]: Needs renaming, not important now.

File: core/server/api/v2/utils/index.js [hunk 5]
```diff
@@ -11,18 +11,8 @@ module.exports = {
...
-     */
     isContentAPI: (frame) => {
-        const context = frame.options && frame.options.context || {};
-        // CASE: An empty context is considered public by the core/server/services/permissions/parse-context.js
-        //       module, replicated here because the context is unparsed until after the permissions layer of the pipeline
-        const isPublic = context.public || Object.keys(context).length === 0;
-
-        // CASE: apiType = 'content' for HTTP Content API
-        return frame.apiType === 'content' || isPublic;
+        return frame.apiType === 'content';
```
Comment (kirrg001) [inline core/server/api/v2/utils/index.js]: Doesn't matter which context you have. If you request content API, you are in content ctrl land. Same for admin.

File: core/server/api/shared/pipeline.js (hunk 1 above)
Comment (gargol) [reply core/server/api/shared/pipeline.js]: As long as `frameOptions` are populated with docName and method that map to these rules: https://github.com/TryGhost/Ghost/blob/master/core/server/web/shared/middlewares/error-handler.js#L96-L106 there should be no effect. The 404 for `read`/`browse` would map to `read/list {resource}` and this translation:
```
"NotFoundError": "Resource not found error, cannot {action}.",
```

File: core/server/api/v2/index.js (hunk 4 above)
Comment (gargol) [reply core/server/api/v2/index.js]: Would be best tackled during https://github.com/TryGhost/Ghost/issues/10106, right?

**Comment by allouis** [comment]
This looks good - I will test A$AP

**Comment by gargol** [comment]
Have tested the part which had to do with changes in error handling, works great and is a lot cleaner than the previous implementation :+1:

File: core/server/api/shared/pipeline.js (hunk 1 above)
Comment (kirrg001) [reply core/server/api/shared/pipeline.js]: perfect 👍

File: core/server/api/v2/index.js (hunk 4 above)
Comment (kirrg001) [reply core/server/api/v2/index.js]: yes perf 👍

**Comment by kirrg001** [comment]
@allouis Merged. Can you please confirm back if you have tested the apiType change? Thanks!

**Pull request merged**